client/internal/httpsender_test.go

```diff
@@ -55,3 +56,45 @@ func TestHTTPSenderRetryForStatusTooManyRequests(t *testing.T) {
 	cancel()
 	srv.Close()
 }
+
+func TestAddTLSConfig(t *testing.T) {
+	sender := NewHTTPSender(&sharedinternal.NopLogger{})
+
+	certificate, err := GenerateCertificate()
+	assert.NoError(t, err)
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{certificate},
+	}
+
+	sender.AddTLSConfig(tlsConfig)
+	assert.Equal(t, sender.client.Transport, &http.Transport{TLSClientConfig: tlsConfig})
+}
+
+func GenerateCertificate() (tls.Certificate, error) {
+
+	certPem := []byte(`-----BEGIN CERTIFICATE-----
+MIIBhTCCASugAwIBAgIQIRi6zePL6mKjOipn+dNuaTAKBggqhkjOPQQDAjASMRAw
+DgYDVQQKEwdBY21lIENvMB4XDTE3MTAyMDE5NDMwNloXDTE4MTAyMDE5NDMwNlow
+EjEQMA4GA1UEChMHQWNtZSBDbzBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABD0d
+7VNhbWvZLWPuj/RtHFjvtJBEwOkhbN/BnnE8rnZR8+sbwnc/KhCk3FhnpHZnQz7B
+5aETbbIgmuvewdjvSBSjYzBhMA4GA1UdDwEB/wQEAwICpDATBgNVHSUEDDAKBggr
+BgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MCkGA1UdEQQiMCCCDmxvY2FsaG9zdDo1
+NDUzgg4xMjcuMC4wLjE6NTQ1MzAKBggqhkjOPQQDAgNIADBFAiEA2zpJEPQyz6/l
+Wf86aX6PepsntZv2GYlA5UpabfT2EZICICpJ5h/iI+i341gBmLiAFQOyTDT+/wQc
+6MF9+Yw1Yy0t
+-----END CERTIFICATE-----`)
+
+	keyPem := []byte(`-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEIIrYSSNQFaA2Hwf1duRSxKtLYX5CB04fSeQ6tF1aY/PuoAoGCCqGSM49
+AwEHoUQDQgAEPR3tU2Fta9ktY+6P9G0cWO+0kETA6SFs38GecTyudlHz6xvCdz8q
+EKTcWGekdmdDPsHloRNtsiCa697B2O9IFA==
+-----END EC PRIVATE KEY-----`)
```

Comment:
Are these a sample cert for testing? Add a comment to clarify.

Reply:
@tigrannajaryan thanks for the quick reply. I have now implemented a new function called _TestConnectWithTLS_ on _clientimpl_test.go_ file. It starts mock server with httptest TLS server and check connection for each transport type. 
Had to refactor few methods, hope it is ok.

Please let me know if this is fine for you or if you have different thought on how to implement it.

Reply:
With regards to the certificate, I got from the example on [crypto/tls library](https://pkg.go.dev/crypto/tls#pkg-index), but I could also build a function that generates a random certificate. What do you think?